ice_setup/ice.py

```diff
@@ -1525,15 +1754,12 @@ def _main(argv=None):
     # parse first with no help; set defaults later
     parser.catch_version = __version__
     parser.catch_help = ice_help()
-    parser.dispatch()
-
-    # if dispatch did not catch anything now, parse help
-    parser.catches_help()
-    parser.catches_version()
+    subcmds = parser.dispatch()
```

Comment:
Yeah, I'm not super happy about this is as a solution, but it works.  Do you see a more elegant way?